src/inverse-kinematics/include/iDynTree/InverseKinematics.h

```diff
@@ -166,6 +167,84 @@ class iDynTree::InverseKinematics
 
     enum InverseKinematicsRotationParametrization rotationParametrization();
 
+    /*! @name Parameters-related methods
+     */
+    ///@{
+
+    /**
+     * Sets Maximum Iteration.
+     *
+     * The default value for this parameter is 3000 .
+     *
+     * @param max_iter exits if iter>=max_iter (max_iter<0
+     *                   disables this check).
+     */
+    void setMaxIter(const int max_iter);
```

Comment:
Actually my design goal were mostly the compatibility with iKinIpOpt , I copied the methods from them. We can also have the key-value stuff, especially if we don't recreate IpOpt Problem at each call to solve, so we can set/get directly the parameters on the IpOpt problem instance as iKinIpOpt does.